ogr/ogrspatialreference.cpp

```diff
@@ -172,8 +178,34 @@ struct OGRSpatialReference::Private
     const char *nullifyTargetKeyIfPossible(const char *pszTargetKey);
 
     void refreshAxisMapping();
+
+    struct OptionalLockGuard
+    {
+        Private &m_private;
+
+        explicit OptionalLockGuard(Private *p) : m_private(*p)
+        {
+            if (m_private.m_bIsThreadSafe)
```

Comment:
changed to
```

/** Assignment method, with thread-safety.
 *
 * Same as an assignment operator, but asking also that the *this instance
 * becomes thread-safe.
 *
 * @param oSource SRS to assign to *this
 * @return *this
 * @since 3.10
 */

OGRSpatialReference &
OGRSpatialReference::AssignAndSetThreadSafe(const OGRSpatialReference &oSource)
{
    *this = oSource;
    d->SetThreadSafe();
    return *this;
}
```